be/src/exec/pipeline/query_context.cpp

```diff
@@ -137,6 +137,11 @@ void QueryContext::init_mem_tracker(int64_t query_mem_limit, MemTracker* parent,
         } else {
             _mem_tracker = std::make_shared<MemTracker>(MemTracker::QUERY, query_mem_limit, _profile->name(), parent);
         }
+
+        _static_query_mem_limit = parent->limit();
+        if (query_mem_limit > 0) {
+            _static_query_mem_limit = std::min(query_mem_limit, _static_query_mem_limit);
```

Comment:
`_static_query_mem_limit` (`parent->limit()`) may be non-positive.
```suggestion
            _static_query_mem_limit = _static_query_mem_limit <= 0 ? query_mem_limit : std::min(query_mem_limit, _static_query_mem_limit);
```

Reply:
dear, which value could be positive for sure? I want this value to be positive any way.